### ISSUE TITLE
app Freeze

### ISSUE THREAD

**Issue by JackQiangXJ** [ChatSecure/ChatSecure-iOS#360]
when in message screen, keep receive message from buddy, app will freeze 1s to 2s.

I think it caused by  [self.collectionview reloaddata] in the func ----- yapDatabaseModified:

**Comment by chrisballinger** [comment]
Does this happen all the time? Do you have a large number of messages? It
may be an issue with a 3rd party framework.

**Comment by JackQiangXJ** [comment]
yes,it happen all the time,I think you can receive 50 offline messages onetime.in my opinion,it caused by when received message,app will get a yapdbnotification in messageviewcontroller, then you will reload all data in this screen.

**Comment by JackQiangXJ** [comment]
yes,it happen all the time,I think you can try to receive 50 offline messages onetime.in my opinion,it caused by when received message,app will get a yapdbnotification in messageviewcontroller, then you will reload all data in this screen.

**Comment by chuquangbao** [comment]
Yeah, I have this same problems. 
And, I have more 2000 messages. App free when i am typing ( send chat and save chat state).
How can i solve it?

**Comment by chrisballinger** [comment]
We should limit the number of messages to some same default in the Yap
mappings, and provide a way to scroll back through your history and load
that dynamically.
